Return proper error message when supplied argument for services directory is not a directory

If the supplied argument for services is not a directory, the user will see the following error message:

```
{"timestamp":"1535391807.733618259","source":"blockhead-broker","message":"blockhead-broker.main","log_level":3,"data":{"error":"readdirent: not a directory","trace":"goroutine 1 [running]:\ngithub.com/cloudfoundry-incubator/blockhead/vendor/code.cloudfoundry.org/lager.(*logger).Fatal(0xc42015a0c0, 0x703ab2, 0x4, 0x73b6e0, 0xc4201462a0, 0x0, 0x0, 0x0)\n\t/home/pivotal/go/src/github.com/cloudfoundry-incubator/blockhead/vendor/code.cloudfoundry.org/lager/logger.go:138 +0xcd\nmain.main()\n\t/home/pivotal/go/src/github.com/cloudfoundry-incubator/blockhead/cmd/broker/main.go:28 +0x3a5\n"}}
panic: readdirent: not a directory

goroutine 1 [running]:
github.com/cloudfoundry-incubator/blockhead/vendor/code.cloudfoundry.org/lager.(*logger).Fatal(0xc42015a0c0, 0x703ab2, 0x4, 0x73b6e0, 0xc4201462a0, 0x0, 0x0, 0x0)
        /home/pivotal/go/src/github.com/cloudfoundry-incubator/blockhead/vendor/code.cloudfoundry.org/lager/logger.go:162 +0x60f
main.main()
        /home/pivotal/go/src/github.com/cloudfoundry-incubator/blockhead/cmd/broker/main.go:28 +0x3a5
```

Let's fix it so that proper error message is shown instead of a panic.

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -66,8 +66,10 @@ func NewState(configPath string, servicePath string) (*State, error) {
 		return nil, fmt.Errorf("Service Directory Missing")
 	}
 
-	if _, err := os.Stat(servicePath); os.IsNotExist(err) {
+	if mode, err := os.Stat(servicePath); os.IsNotExist(err) {
 		return nil, fmt.Errorf("%v: %s", err, servicePath)
+	} else if !mode.IsDir() {
+		return nil, fmt.Errorf("A service directory is expected as the second argument.")
 	}
 
 	files, err := ioutil.ReadDir(servicePath)

--- a/pkg/config/config_test.go
+++ b/pkg/config/config_test.go
@@ -79,6 +79,14 @@ var _ = Describe("Config", func() {
 				})
 			})
 
+			Context("when service directory argument is not a directory", func() {
+				It("errors when services dir is missing", func() {
+					_, err := config.NewState(configPath, configPath)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("A service directory is expected as the second argument."))
+				})
+			})
+
 			Context("when service directory exists", func() {
 				BeforeEach(func() {
 					servicePath, err = ioutil.TempDir("", "service-dir")

